### PR TITLE
[GOBBLIN-1951] Emit GTE when deleting corrupted ORC files

### PR DESCRIPTION
**Pull request by homatthew**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-1951] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1951

### Description
- [X] Here are some details about my PR, including screenshots (if applicable):

This commit modifies ORC file validation during the commit phase and deletes corrupted files to also emit a GTE. It also includes a test for ORC file validation.

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
image

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/InstrumentedGobblinOrcWriter.java [hunk 1]
```diff
@@ -39,18 +37,16 @@
  */
 @Slf4j
 public class InstrumentedGobblinOrcWriter extends GobblinOrcWriter {
-  MetricContext metricContext;
```
Comment (homatthew) [inline gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/InstrumentedGobblinOrcWriter.java:42]: Moved to the base class

**Review by ZihanLi58 — COMMENTED** [review]
Build/test fails, please take a look

File: gobblin-modules/gobblin-orc/src/test/java/org/apache/gobblin/writer/GobblinBaseOrcWriterTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    OrcFile.ReaderOptions readerOptions = new OrcFile.ReaderOptions(conf);
+
+    MetricContext mockContext = Mockito.mock(MetricContext.class);
+    Path p = new Path(corruptedOrcFile.getAbsolutePath());
+    Assert.assertThrows(FileFormatException.class,
+        () -> GobblinBaseOrcWriter.assertOrcFileIsValid(fs, p, readerOptions, mockContext));
+
+    GobblinEventBuilder eventBuilder = new GobblinEventBuilder(CORRUPTED_ORC_FILE_DELETION_EVENT, GobblinBaseOrcWriter.ORC_WRITER_NAMESPACE);
```
Comment (hanghangliu) [inline gobblin-modules/gobblin-orc/src/test/java/org/apache/gobblin/writer/GobblinBaseOrcWriterTest.java:62]: Do you think test the EventBuilder like [this](https://github.com/apache/gobblin/blob/10397ab9d85829b86f1a04958742e612dcff234b/gobblin-iceberg/src/test/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriterTest.java#L179) makes it easier so you won't need to suppress the warning?

Comment (homatthew) [reply gobblin-modules/gobblin-orc/src/test/java/org/apache/gobblin/writer/GobblinBaseOrcWriterTest.java:62]: Do you think the warning was coming from the tests? The warning came from the impl and not the test.

**Review by ZihanLi58 — APPROVED** [review]
+1

**Pull request merged**